lang/test/org/partiql/lang/errors/ParserErrorsTest.kt

```diff
@@ -2110,6 +2110,56 @@ class ParserErrorsTest : SqlParserTestBase() {
             Property.TOKEN_TYPE to TokenType.IDENTIFIER,
             Property.TOKEN_VALUE to ion.newSymbol("undrop")))
 
+    @Test
+    fun execWithDmlInsert() = checkInputThrowingParserException(
+        "EXEC abc_proc INSERT INTO dogs <<{}>>",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 15L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("insert_into")))
+
+    @Test
+    fun execWithDmlDelete() = checkInputThrowingParserException(
+        "EXEC abc_proc DELETE FROM dogs WHERE name = 'Jack'",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 15L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("delete")))
+
+    @Test
+    fun execWithDmlUpdate() = checkInputThrowingParserException(
+        "EXEC abc_proc UPDATE dogs SET dogs.name = '1'",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 15L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("update")))
+
+    @Test
+    fun execWithDmlRemove() = checkInputThrowingParserException(
+        "EXEC abc_proc FROM dogs REMOVE dogs.name",
+        ErrorCode.PARSE_UNEXPECTED_TERM,
+        mapOf(
+            Property.LINE_NUMBER to 1L,
+            Property.COLUMN_NUMBER to 25L,
+            Property.TOKEN_TYPE to TokenType.KEYWORD,
+            Property.TOKEN_VALUE to ion.newSymbol("remove")))
```

Comment:
Ideally, I think the error should point to the `FROM` on column 15. `FROM` is not currently denoted as a "topLevelType" ([source](https://github.com/partiql/partiql-lang-kotlin/blob/1462dd351fa7dbba0493b6b2d9e87af8a0b3ac3c/lang/src/org/partiql/lang/syntax/SqlParser.kt#L128)), so this may be where the issue comes from.